annotation-utils/src/main/java/edu/isi/vista/annotationutils/RestoreJson.kt

```diff
@@ -1,109 +1,149 @@
+ * https://github.com/isi-vista/nlp-util/blob/master/common-core-open/src/main/java/edu/isi/nlp/parameters/serifstyle/SerifStyleParameterFileLoader.java
+ *
+ * An example is provided in curated-training-annotator/sample_params/restore-text.params
+ * The required parameter values are:
+ * <ul>
+ *     <li> {@code indexDirectory} is the location of the files produced by {@code IndexFlatGigaword.java}
+(https://github.com/isi-vista/nlp-util/blob/master/nlp-core-open/src/main/java/edu/isi/nlp/corpora/gigaword/IndexFlatGigaword.java) </li>
+ *     <li> {@code gigawordDataDirectory} is the location of the gigaword text files </li>
+ *     <li> {@code inputJsonDirectory} is the location of the stripped json files produced by {@code ExportAnnotations.kt} </li>
+ *     <li> {@code restoredJsonDirectory} is where the new json files will go </li>
+ *  </ul>
+ */
+
+class RestoreJson {
+    companion object {
+        fun main(argv: Array<String>) {
+            if (argv.size != 1) {
+                throw RuntimeException("Expected a single argument, a parameter file")
+            }
+            val params = Parameters.loadSerifStyle(File(argv[0]))
+            restore(params)
+        }
+        fun restore(params: Parameters) {
+            val inputJsonDirectory = params.getExistingDirectory("inputJsonDirectory")!!
+            val outputDirectory = params.getCreatableDirectory("restoredJsonDirectory")!!
+
+            // Make objects for reading, parsing, and writing json:
+            val objectMapper = ObjectMapper()
+            val prettyPrinter = objectMapper.writerWithDefaultPrettyPrinter()
+            val gigaWordTextSource = makeTextSource(params, 'gigaword')
+            val aceTextSource = makeTextSource(params, 'ace')
+
+            inputJsonDirectory.walk().filter { it.isFile }.forEach { jsonFile ->
+                val filename = jsonFile.name
+                //returns docID if ace file, else returns an empty string
+                val aceDocID = getAceDocID(filename)
+                if (filename.contains(Regex("[\b_]ENG[\b_]")) || aceDocID.isNotEmpty()) {
+                    // The project directory is the path to this file with the input directory components
+                    // stripped off the front, then joined with the desired output directory.
+                    val projectOutDir = Paths.get(
+                            outputDirectory.toString(),
+                            jsonFile.toPath().removePrefixPath(inputJsonDirectory.toPath()).toString()
+                    ).parent.toAbsolutePath()
+                    if (Files.notExists(projectOutDir)) {
+                        Files.createDirectories(projectOutDir)
+                    }
+                    val jsonTree = objectMapper.readTree(jsonFile)
+                    var text = ""
+                    // If aceDocID is empty, it means the document being processed is not an
+                    // ace document
+                    if(aceDocID.isNotEmpty()) {

```

Comment:
Instead of initializing `text` before this if-statement, you can do something like this:
```
val text = if (aceDocID.isNotEmpty()) {
  ...
  aceTextSource.getOriginalText(docID).orNull()
    ?: throw RuntimeException("Could not get original text for $docID")
} else {
  ...
```